compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1047,6 +1049,23 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
             ) if ns == ValueNS => {
                 bad_struct_syntax_suggestion(def_id);
             }
+            (Res::Def(DefKind::Ctor(_, CtorKind::Const), def_id), _) if ns == ValueNS => {
+                match source {
+                    PathSource::Expr(_) | PathSource::TupleStruct(..) | PathSource::Pat => {
+                        let span = find_span(&source, err);
+                        if let Some(span) = self.def_span(def_id) {
+                            err.span_label(span, &format!("`{}` defined here", path_str));
+                        }
+                        err.span_suggestion(
+                            span,
+                            &format!("use this syntax instead"),
+                            format!("{path_str}"),
+                            Applicability::MaybeIncorrect,
```

Comment:
Shouldn't this be machine applicable? In what cases it will be incorrect?

Reply:
I mean, there are two different suggestions that get emitted, so there *could* be some ambiguity. (I guess the "help: a tuple struct with a similar name exists" suggestion *might* just never be what the user wants, but that decision would have to be made in another PR)